crates/hyle-model/src/contract.rs

```diff
@@ -757,15 +746,30 @@ impl Add<BlockHeight> for BlockHeight {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+#[cfg_attr(feature = "full", derive(utoipa::ToSchema))]
+pub enum TimeoutWindow {
+    NoTimeout,
+    Timeout(BlockHeight),
+}
+
+impl Default for TimeoutWindow {
+    fn default() -> Self {
+        TimeoutWindow::Timeout(BlockHeight(100))
+    }
+}
+
 #[derive(
     Debug, Serialize, Deserialize, Default, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize,
 )]
+#[cfg_attr(feature = "full", derive(utoipa::ToSchema))]
 /// Used as a blob action to register a contract.
 pub struct RegisterContractAction {
     pub verifier: Verifier,
     pub program_id: ProgramId,
     pub state_commitment: StateCommitment,
     pub contract_name: ContractName,
+    pub timeout_window: Option<TimeoutWindow>,
```

Comment:
is it possible to have a bit of rust doc on this property ? It's unclear the difference between None and Timeout(0)

Reply:
If None, it either won't update the current value or will use the default. Otherwise it willset the value to whatever is passed.